media/js/mozorg/plugincheck-update.js

```diff
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function($) {
+    'use strict';
+
+    var client = window.Mozilla.Client;
+    var body = $('body');
+
+    if (client._isFirefoxDesktop) {
+        client.getFirefoxDetails(function(details) {
+            if (details.isUpToDate) {
```

Comment:
All good questions. Pre-release browsers seem to report as "up to date" even if they're not up to date on their own channel (e.g. I haven't restarted Nightly in four days but the page tells me it's up to date). We initially decided not to do anything special for pre-release channels because this page is mostly aimed at old Firefoxes that still have the in-product link. But then... if you're on a year old copy of Dev Edition that should still report as outdated, but you'll get a download button for release channel... I'm not sure what to do about that scenario.

We do probably need to do something special for ESR. I agree that if someone is on an old ESR they should probably update to the latest ESR, not release. We can use the same messaging for ESR, we just need to give them the right download. Does the button logic already do this?